### ISSUE TITLE
Crash with Collector Level 4 on Level 13 Normal

### ISSUE THREAD

**Issue by codingthat** [larsiusprime/tdrpg-bugs#1656]
I wanted to try out the ball and chain, so I edited my save file to add some XP, levels, and skill points to the Collector.

I noticed that spending a handful of points then made the points suddenly go to zero.

Nonetheless, I went to Level 13, placed a Collector, boosted to Level 4 (highest I could), and put time to 4X.  After a second or two [it crashed](https://github.com/larsiusprime/tdrpg-bugs/files/4007927/DQ2_2019-12-29_10.25.53_CRASH.zip) with

```
com.leveluplabs.tdrpg.CreatureSprite.removeThorns (com/leveluplabs/tdrpg/CreatureSprite.hx line 3448)
com.leveluplabs.tdrpg.CreatureSprite.removeStatusEffect (com/leveluplabs/tdrpg/CreatureSprite.hx line 1385)
com.leveluplabs.tdrpg.CreatureSprite.updateStatus (com/leveluplabs/tdrpg/CreatureSprite.hx line 1181)
com.leveluplabs.tdrpg.CreatureSprite.updateFromModel (com/leveluplabs/tdrpg/CreatureSprite.hx line 2993)
com.leveluplabs.tdrpg.DefenderSprite.updateFromModel (com/leveluplabs/tdrpg/DefenderSprite.hx line 1655)
com.leveluplabs.tdrpg.BattleModel.updateDefenders (com/leveluplabs/tdrpg/BattleModel.hx line 4255)
com.leveluplabs.tdrpg.BattleModel.doUpdate (com/leveluplabs/tdrpg/BattleModel.hx line 2389)
com.leveluplabs.tdrpg.BattleModel.update (com/leveluplabs/tdrpg/BattleModel.hx line 1946)
com.leveluplabs.tdrpg.State_Battle.battleUpdate (com/leveluplabs/tdrpg/State_Battle.hx line 1992)
com.leveluplabs.tdrpg.State_Battle.update (com/leveluplabs/tdrpg/State_Battle.hx line 1716)
flixel.FlxState.tryUpdate (flixel/FlxState.hx line 172)
flixel.FlxGame.update (flixel/FlxGame.hx line 738)
com.leveluplabs.tdrpg.Main.update (com/leveluplabs/tdrpg/Main.hx line 2391)
flixel.FlxGame.step (flixel/FlxGame.hx line 668)
flixel.FlxGame.onEnterFrame (flixel/FlxGame.hx line 538)
openfl.events.EventDispatcher.__dispatchEvent (openfl/events/EventDispatcher.hx line 443)
openfl.display.DisplayObject.__dispatch (openfl/display/DisplayObject.hx line 1234)
```

Completely reproducible.  Happens a moment after the first attacker is killed.
* * *

DQ2 Universal 0.0.6 native on Linux Mint 19 (64-bit) in full-screen mode on a 1366x768 single-monitor setup with an Intel HD Graphics 4400 chipset.

**Comment by codingthat** [comment]
Also happens if I only edit the save file to help bring the Collector to Level 10.  Seems related to Replenish, because if I only boost to Level 3 in battle, nothing happens.  Boost to 4 and it crashes.
* * *

DQ2 Universal 0.0.6 native on Linux Mint 19 (64-bit) in 1280x720 windowed mode on a 1366x768 single-monitor setup with an Intel HD Graphics 4400 chipset.

**Comment by larsiusprime** [comment]
Thanks for posting the crash log! I grabbed the save file and wasn't able to reproduce this unfortunately. From the looks of it, whatever this was caused by is related to the item your collector is wearing, which grants thorns. You'll notice the stack trace is referencing thorns.

Although I couldn't get a clean repro, I found the line that was crashing and threw in some null checks. Maybe that will solve it? We'll see again in 0.0.7. 

Closing for now, let's reopen if it happens again after 0.0.7.
